gtclang/test/atlas-integration-test/GenerateUnstructuredStencils.cpp

```diff
@@ -0,0 +1,139 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                         _       _
+//                        | |     | |
+//                    __ _| |_ ___| | __ _ _ __   __ _
+//                   / _` | __/ __| |/ _` | '_ \ / _` |
+//                  | (_| | || (__| | (_| | | | | (_| |
+//                   \__, |\__\___|_|\__,_|_| |_|\__, | - GridTools Clang DSL
+//                    __/ |                       __/ |
+//                   |___/                       |___/
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/CodeGen/CXXNaive-ico/CXXNaiveCodeGen.h"
+#include "dawn/Unittest/IIRBuilder.h"
+
+#include <cstring>
+#include <fstream>
+
+template <typename CG>
+void dump(std::ostream& os, dawn::codegen::stencilInstantiationContext& ctx) {
+  dawn::DiagnosticsEngine diagnostics;
+  CG generator(ctx, diagnostics, 0);
+  auto tu = generator.generateCode();
+
+  std::ostringstream ss;
+  for(auto const& macroDefine : tu->getPPDefines())
+    ss << macroDefine << "\n";
+
+  ss << tu->getGlobals();
+  for(auto const& s : tu->getStencils())
+    ss << s.second;
+  os << ss.str();
+}
+
+int main() {
+
+  {
+    using namespace dawn::iir;
+    using LocType = dawn::ast::Expr::LocationType;
+
+    UnstructuredIIRBuilder b;
+    auto in_f = b.field("in_field", LocType::Cells);
+    auto out_f = b.field("out_field", LocType::Cells);
+
+    auto stencil_instantiation = b.build(
```

Comment:
why did you chose to test the code-gen only and not have unstructured SIR and run it trough the full toolchain?